src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.cs

```diff
@@ -113,6 +113,21 @@ internal TarHeader(TarEntryFormat format, TarEntryType typeFlag, TarHeader other
         internal void InitializeExtendedAttributesWithExisting(IEnumerable<KeyValuePair<string, string>> existing)
         {
             Debug.Assert(_ea == null);
+            foreach (KeyValuePair<string, string> kvp in existing)
+            {
+                if (kvp.Key.IndexOf('=') != -1)
+                {
+                    throw new ArgumentException(SR.Format(SR.TarExtAttrDisallowedKeyChar, kvp.Key, '='));
+                }
+                if (kvp.Key.IndexOf('\n') != -1)
+                {
+                    throw new ArgumentException(SR.Format(SR.TarExtAttrDisallowedKeyChar, kvp.Key, "\\n"));
+                }
```

Comment:
```suggestion
                int index = kvp.Key.IndexOfAny('=', '\n');
                if (index >= 0)
                {
                    throw new ArgumentException(SR.Format(SR.TarExtAttrDisallowedKeyChar, kvp.Key, kvp.Key[index]));
                }
                if (kvp.Key.IndexOf('\n') >= 0)
                {
                    throw new ArgumentException(SR.Format(SR.TarExtAttrDisallowedKeyChar, kvp.Key, "\\n"));
                }
```

Reply:
Thanks. I am seeing that if I use `kvp.Key[index]` as suggested in line 121, thew exception message seen by the user will show an actual new line, instead of literally `\n`. It's why I had opted to use "\\n" originally.

So I can do this instead:

```cs
throw new ArgumentException(SR.Format(SR.TarExtAttrDisallowedKeyChar, kvp.Key, 
kvp.Key[index] == '\n' ? "\\n" : kvp.Key[index]));
```

Reply:
> So I can do this instead

Yup